src/arrayMethodReduce.js

```diff
@@ -6,6 +6,19 @@
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
     // write code here
+    let startValue = initialValue;
+
+    if (arguments.length < 2) {
+      startValue = this[0];
+    }
+
+    const startIndex = (arguments.length < 2) ? 1 : 0;
```

Comment:
you can transform true into 1 and false into 0 right away
```suggestion
    const startIndex = Number(arguments.length < 2);
```